### PR TITLE
Qt1-176 rpl summary endpoints

### PR DESCRIPTION
**Pull request by Pauljgraham**

### CODE REVIEW

**Review by alireid — DISMISSED** [review]
Code run through and working end to end performed. All good.

File: src/Approvals/SFA.DAS.Approvals.Api/Controllers/DraftApprenticeshipController.cs [hunk 1]
```diff
@@ -236,5 +239,49 @@ public async Task<IActionResult> UpdateDraftApprenticeship(long cohortId, long a
...
+            {
+                return NotFound();
+            }
+
+            return Ok(result);
+        }
+
+        [HttpPost]
+        [Route("provider/{providerId}/unapproved/{cohortId}/apprentices/{draftApprenticeshipId}/edit/prior-learning-data")]
+        public async Task<IActionResult> PriorLearningData(long cohortId, long draftApprenticeshipId, [FromBody] InnerApi.Requests.AddPriorLearningDataRequest request)
```
Comment (chrisfoster76) [inline src/Approvals/SFA.DAS.Approvals.Api/Controllers/DraftApprenticeshipController.cs:259]: An inner api request should not be a parameter on an outer api endpoint.

File: src/Approvals/SFA.DAS.Approvals/Application/DraftApprenticeships/Commands/AddPriorLearningData/AddPriorLearningDataCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                IsDurationReducedByRpl = request.IsDurationReducedByRpl,
+                DurationReducedBy = request.DurationReducedBy,
+                CostBeforeRpl = request.CostBeforeRpl,
+                PriceReducedBy = request.PriceReducedBy
+            };
+            var result = await _apiClient.PostWithResponseCode<AddPriorLearningDataResponse>(new PostAddPriorLearningDataRequest(request.CohortId, request.DraftApprenticeshipId, addPriorLearningRequest), false);
+
+            result.EnsureSuccessStatusCode();
+
+            var apprenticeship = await _apiClient.Get<GetDraftApprenticeshipResponse>(new GetDraftApprenticeshipRequest(request.CohortId, request.DraftApprenticeshipId));
```
Comment (chrisfoster76) [inline src/Approvals/SFA.DAS.Approvals/Application/DraftApprenticeships/Commands/AddPriorLearningData/AddPriorLearningDataCommandHandler.cs:39]: You could await both of these tasks together by the looks of it.

File: src/Approvals/SFA.DAS.Approvals/Application/DraftApprenticeships/Queries/GetEditDraftApprenticeshipPriorLearningSummary/GetEditDraftApprenticeshipPriorLearningSummaryQueryHandler.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+        public GetEditDraftApprenticeshipPriorLearningSummaryQueryHandler(ICommitmentsV2ApiClient<CommitmentsV2ApiConfiguration> apiClient)
+        {
+            _apiClient = apiClient;
+        }
+
+        public async  Task<GetEditDraftApprenticeshipPriorLearningSummaryQueryResult> Handle(GetEditDraftApprenticeshipPriorLearningSummaryQuery request, CancellationToken cancellationToken)
+        {
+            var apprenticeship = await _apiClient.Get<GetDraftApprenticeshipResponse>(new GetDraftApprenticeshipRequest(request.CohortId, request.DraftApprenticeshipId));
+            var rplSummary = await _apiClient.Get<GetPriorLearningSummaryResponse>(new GetPriorLearningSummaryRequest(request.CohortId, request.DraftApprenticeshipId));
```
Comment (chrisfoster76) [inline src/Approvals/SFA.DAS.Approvals/Application/DraftApprenticeships/Queries/GetEditDraftApprenticeshipPriorLearningSummary/GetEditDraftApprenticeshipPriorLearningSummaryQueryHandler.cs:23]: You could await these tasks together.

**Review by chrisfoster76 — COMMENTED** [review]
Minor things

File: src/Approvals/SFA.DAS.Approvals.Api/Controllers/DraftApprenticeshipController.cs (hunk 1 above)
Comment (Pauljgraham) [reply src/Approvals/SFA.DAS.Approvals.Api/Controllers/DraftApprenticeshipController.cs:259]: fixed

File: src/Approvals/SFA.DAS.Approvals/Application/DraftApprenticeships/Queries/GetEditDraftApprenticeshipPriorLearningSummary/GetEditDraftApprenticeshipPriorLearningSummaryQueryHandler.cs (hunk 3 above)
Comment (Pauljgraham) [reply src/Approvals/SFA.DAS.Approvals/Application/DraftApprenticeships/Queries/GetEditDraftApprenticeshipPriorLearningSummary/GetEditDraftApprenticeshipPriorLearningSummaryQueryHandler.cs:23]: done

**Review by chrisfoster76 — APPROVED** [review]

**Pull request merged**